### PR TITLE
Require the strict types declaration

### LINKED ISSUE #679: Adopt strict typing
Now that we no longer support PHP 5 we can start using strict typing. This will help make our code more reliable since it will prevent us from doing any unwanted type casting.

In order to enforce this, we can require `slevomat/coding-standard` and add the following declaration to our `phpcs-ruleset.xml.dist`:

```
    <!-- Require the strict types declaration in every PHP file. -->
    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
        <properties>
            <property name="newlinesCountBetweenOpenTagAndDeclare" value="2"/>
        </properties>
    </rule>
```

### PR DESCRIPTION
**Pull request by MPParsley**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     },
     "require-dev": {
         "drupal/coder": "^8.2",
-        "phpunit/phpunit": "^7 || ^8"
+        "phpunit/phpunit": "^7 || ^8",
+		"slevomat/coding-standard": "~6.0"
```
Comment (MPParsley) [inline composer.json]:
```suggestion
        "slevomat/coding-standard": "~6.0"
```

File: phpcs-ruleset.xml.dist [hunk 2]
```diff
@@ -20,5 +20,13 @@
         <!-- Deprecation versions are allowed to include beta versions. -->
         <exclude name="Drupal.Semantics.FunctionTriggerError.TriggerErrorVersion" />
     </rule>
+
     <rule ref="./vendor/drupal/coder/coder_sniffer/DrupalPractice"></rule>
+
+    <!-- Require the strict types declaration in every PHP file. -->
+    <rule ref="SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
```
Comment (MPParsley) [inline phpcs-ruleset.xml.dist:28]:
```suggestion
    <rule ref="./vendor/slevomat/coding-standard/SlevomatCodingStandard/SlevomatCodingStandard.TypeHints.DeclareStrictTypes">
```

**Comment by MPParsley** [comment]
I suggest we define the types for all interfaces in follow up issues as this would become much too large for one PR.

File: tests/src/Unit/GroupSubscribeFormatterTest.php [hunk 3]
```diff
@@ -308,7 +310,7 @@ public function testNoSubscribePermission() {
     $this->accessResult->isAllowed()->willReturn(FALSE);
 
     $elements = $this->getElements();
-    $this->assertTrue(strpos($elements[0]['#value'], 'This is a closed group.') === 0);
+    $this->assertStringStartsWith('This is a closed group.', $elements[0]['#value']->render());
```
Comment (pfrenssen) [inline tests/src/Unit/GroupSubscribeFormatterTest.php:313]: :+1:

**Review by pfrenssen — APPROVED** [review]

**Pull request merged**